alot/settings/manager.py

```diff
@@ -77,13 +77,21 @@ def read_config(self, path):
         self._config.merge(newconfig)
         self._config.walk(self._expand_config_values)
 
-        hooks_path = os.path.expanduser(self._config.get('hooksfile'))
-        try:
-            spec = importlib.util.spec_from_file_location('hooks', hooks_path)
-            self.hooks = importlib.util.module_from_spec(spec)
-            spec.loader.exec_module(self.hooks)
-        except:
-            logging.exception('unable to load hooks file:%s', hooks_path)
+        # set up hooks module if requested
+        hooks_path = self._config.get('hooksfile')
+        if hooks_path:
+            hooks_path = os.path.expanduser(hooks_path)
+            logging.debug('hooks: loading from: %s', hooks_path)
+            try:
+                spec = importlib.util.spec_from_file_location('hooks',
+                                                              hooks_path)
+                self.hooks = importlib.util.module_from_spec(spec)
+                spec.loader.exec_module(self.hooks)
+            except:
+                logging.exception('unable to load hooks file:%s', hooks_path)
+        else:
+            logging.debug('hooks: hooksfile config option is unset')
```

Comment:
For a better transition for users we could add a compatibility code path here. If the value in the config file is `None` (unset)  we would at this point issue a deprecation warning and set the value to the old default value and try to load it.  We could also announce a version until which this extra code path will exist.

The user would then have to set some value like `None` in the config file explicitly in order to silence the message and get the new behaviour (no hooks file loaded).  Is it possible for the python code to differentiate between an implicit and explicit `None`?

Reply:
> For a better transition for users we could add a compatibility code path here. If the value in the config file is `None` (unset) we would at this point issue a deprecation warning 

That's why I added this extra line reporting on the unset value.

> ...and set the value to the old default value and try to load it. We could also announce a version until which this extra code path will exist.

This would mean we had to adjust the tests which fail simply because a missing hooks file (at the default path) raises an exception. Not so nice. 

> The user would then have to set some value like `None` in the config file explicitly in order to silence the message and get the new behaviour (no hooks file loaded). Is it possible for the python code to differentiate between an implicit and explicit `None`?

I don't think this distinction is possible. What one could do is check if the config contains the key "hooksfile" or not, but I am not sure how do do this in combination with the typechecking that's going on for the config.

Reply:
As I said I don't care much if this is a breaking change because I follow alot's development quite closely.  

It might be problematic for users who expect the old behaviour and are at a loss why their hooks file does not work.  Such a user might not even run alot with debugging enabled so they might not see the message.  But if we raise the level to "error" or similar we need a way to silence it.  Maybe we can advertise putting `hooksfile = /dev/null`  into the config.

Reply:
> As I said I don't care much if this is a breaking change because I follow alot's development quite closely.

Sorry, I misunderstood you there. right.

> It might be problematic for users who expect the old behaviour and are at a loss why their hooks file does not work. Such a user might not even run alot with debugging enabled so they might not see the message. 

yes, agreed.

> But if we raise the level to "error" or similar we need a way to silence it.

yes, I don't think that's a brilliant idea either. We could temporarily have a notification upon start, but that may be distracting to new users who actually haven't configured hooks yet.

We can certainly add a warning to the docs (see my commit) but also mention the change in the changelog and release announcement. For me that'd be enough really.


> Maybe we can advertise putting `hooksfile = /dev/null` into the config.

Wouldn't that trigger errors as it'd made alot try to parse that file, which should fail?